packages/form-core/src/NativeTextFieldMixin.js

```diff
@@ -90,27 +96,33 @@ const NativeTextFieldMixinImplementation = superclass =>
      * @protected
      */
     _setValueAndPreserveCaret(newValue) {
-      // Only preserve caret if focused (changing selectionStart will move focus in Safari)
-      if (this.focused) {
-        // Not all elements might have selection, and even if they have the
-        // right properties, accessing them might throw an exception (like for
-        // <input type=number>)
-        try {
-          // SelectElement doesn't have selectionStart/selectionEnd
-          if (!(this._inputNode instanceof HTMLSelectElement)) {
-            const start = this._inputNode.selectionStart;
+      // Only set if newValue is new, fix for Safari bug: https://github.com/ing-bank/lion/issues/1415
+      if (this._inputNode.value !== newValue) {
```

Comment:
Maybe move to the top of the function to make the nexted if structure more readable:
```js
if (this._inputNode.value === newValue) { 
  return;
}
```

Reply:
The return made lots of tests of the `textarea` fail, i now moved the if statement one function up.